includes/class-wcpdf-main.php

```diff
@@ -316,20 +316,21 @@ public function generate_pdf_ajax() {
 			}
 
 			// Check the user privileges
-			if( !( current_user_can( 'manage_woocommerce_orders' ) || current_user_can( 'edit_shop_orders' ) ) && !isset( $_GET['my-account'] ) ) {
-				$allowed = false;
-			}
-
-			// User call from my-account page
-			if ( !current_user_can('manage_options') && isset( $_GET['my-account'] ) ) {
-				// Only for single orders!
-				if ( count( $order_ids ) > 1 ) {
-					$allowed = false;
-				}
-
-				// Check if current user is owner of order IMPORTANT!!!
-				if ( ! current_user_can( 'view_order', $order_ids[0] ) ) {
+			$admin = new Admin();
```

Comment:
admin shouldn't be instantiated twice because this will lead to action hooks and filters being run twice too. Instead you should make the one instance accessible here: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/72d2a02369358f966d252f5583fdeb1f51ea9926/woocommerce-pdf-invoices-packingslips.php#L192

Reply:
Fixed here: https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/pull/356/commits/20dd846321181d69cccacdb66ad28537826ee37b